scipy/linalg/misc.py

```diff
@@ -141,6 +142,13 @@ def norm(a, ord=None, axis=None, keepdims=False, check_finite=True):
     else:
         a = np.asarray(a)
 
+    # define empty array norm as 0.0 for ord > 0
+    if a.size == 0:
+        if ord in (None, "fro") or ord > 0:
+            return 0.0
```

Comment:
Seems to be consistent with `np.linalg.norm([])`, as you say. Perhaps once a linalg expert weighs in the behavior might be encoded in a regression test as well.

Reply:
Argh, `scipy.linalg.norm` doesn't seem to have any tests as far as I can tell, it's only used inside tests for other functionality.

@ogauthe thanks for working on this. Could you add a `scipy/linalg/tests/test_misc.py` file, and add one regression test for this size-0 behavior to it?

Reply:
Returning the scalar 0 is not correct if the shape of `a` is nontrivial and an `axis` has been given.  For example, if `z` has shape `(3, 0, 2)`, and `axis=1` is given, the result should be an array with shape `(3, 2)` if `keepdims` is False.  If `keepdims` is True, the shape should be `(3, 1, 2)`.  E.g.

```
In [95]: z = np.empty((3, 0, 2))

In [96]: z
Out[96]: array([], shape=(3, 0, 2), dtype=float64)

In [97]: z.size
Out[97]: 0

In [98]: np.linalg.norm(z, axis=1)
Out[98]: 
array([[0., 0.],
       [0., 0.],
       [0., 0.]])

In [99]: np.linalg.norm(z, axis=1, keepdims=True)
Out[99]: 
array([[[0., 0.]],

       [[0., 0.]],

       [[0., 0.]]])
```